transports/http/helpers/error_response_helpers.go

```diff
@@ -0,0 +1,48 @@
+package helpers
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/bitcoin-sv/block-headers-service/domains"
+)
+
+// ResponseError is an object that we can return to the client if any error happens
+// on the server
+type ResponseError struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+}
+
+// ErrorResponse is creating an error object to send it to the client
+func ErrorResponse(err error, statusCode int) ResponseError {
```

Comment:
```suggestion
func ErrorResponse(err error, statusCode int) *ResponseError {
```

To avoid copying each time, the standard approach is just use the pointers.
Don't bother the memory - the cool garbage collector would handle this 🙈

Reply:
Please read how we did similar functionality in `spv-wallet`.
https://github.com/bitcoin-sv/spv-wallet/blob/main/models/errors.go
https://github.com/bitcoin-sv/spv-wallet/tree/main/engine/spverrors (especially `http_response.go`)